internal/typeinfo/scan.go

```diff
@@ -0,0 +1,25 @@
+package typeinfo
+
+import "reflect"
+
+// ScanProxy is a shim for scanning query results
+// into types for which we have information.
```

Comment:
The last part of the sentence is not very clear, maybe add that `ScanProxy` is used when scanning directly into the target is not possible/desired